### PR TITLE
Data Types, Operators and Expressions

### LINKED ISSUE #33: Data Types, Operators and Expressions
N | 问题描述 | 输出格式
----|------|------
1 | 交换2个数x和y，用3+种不同方法来实现 | `{x:d} {y:d} => {:d} {:d}`
2 | 判断某个年份year是否为润年 | `{year:d} => {:d}` 
3 | 判断2个浮点数a和b是否相等 | `{a:f} {b:f} {dt:f} => {:d}`
4 | 判断一个整数n是否为偶数 (even) | `{n:d} => {:d}` 
5 | n为一整数，不用乘除及取模求nx17、n%8、n/16 | `{n:d} => {:d} {:d} {:d}`
6 | 判断一个整数n是否为2<sup>k</sup>(2的k次方) | `{n:d} => {:d} `
7 | 位运算分别实现`chmod 600/+x/u+w/go-w` 4种操作 | `{m:o} => {:o} {:o} {:o} {:o}`

**相应说明**:
1. 同时在PR页面描述各方法的优缺点
1. 输出部为"1/0"，表示“True/False”
1. 输出部为"1/0"，表示“相等/不等”
1. 输出部为"1/0"，表示"even/odd"。要求分别用算术运算符和位运算符实现
1. (无)
1. 输出部为"1/0"，表示“True/False”。n要求覆盖“大于、小于、等于”0三种情况
1. 输出部4个new m均基于m初始值计算而得

Fork #20

### PR DESCRIPTION
**Pull request by Nick-Qu**
Resolve #33

Signed-off-by: Nick Qu <nick.quy@qq.com>

### CODE REVIEW

**Comment by Nick-Qu** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 11 22 => 22 11 | {} {} => {} {} | True
2 | 11 22 => 22 11 | {} {} => {} {} | True
3 | 11 22 => 22 11 | {} {} => {} {} | True
4 | 3.14159 3.14156 0.0001 eq | {} {} {} {} | True
5 | 2020 yes | {} {} | True
6 | 5 odd | {} {} | True
7 | 6 even | {} {} | True
8 | 16 yes | {} {} | True
9 | 16 x 17 = 272 | {} x 17 = {} | True
10 | 18 % 8 = 2 | {} % 8 = {} | True
11 | 128 / 16 = 8 | {} / 16 = {} | True
12 | 666: 600 777 666 644 | {}: {} {} {} {} | True

File: operators/nick-qu.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+a = a ^ b
+print(s + " => {} {}".format(a, b))
+
+a = 3.14159
+b = 3.14156
+dt = 1e-4
+if a > b - dt and a < b + dt:
+    print("{} {} {} {}".format(a, b, dt, "eq"))
+else:
+    print("{} {} {} {}".format(a, b, dt, "ne"))
```
Comment (conke) [inline operators/nick-qu.py]: 这种情况不适合2处调用print（下同）

Comment (conke) [inline operators/nick-qu.py]: 这种情况不适合2处调用print（下同）

Comment (conke) [inline operators/nick-qu.py]: 这种情况不适合2处调用print（下同）

**Comment by Nick-Qu** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 11 22 => 22 11 | {} {} => {} {} | True
2 | 11 22 => 22 11 | {} {} => {} {} | True
3 | 11 22 => 22 11 | {} {} => {} {} | True
4 | 3.14159 3.14156 1e-05 ne | {} {} {} {} | True
5 | 2020 yes | {} {} | True
6 | 5 odd | {} {} | True
7 | 6 even | {} {} | True
8 | 16 yes | {} {} | True
9 | 16 x 17 = 272 | {} x 17 = {} | True
10 | 18 % 8 = 2 | {} % 8 = {} | True
11 | 128 / 16 = 8 | {} / 16 = {} | True
12 | 666: 600 777 666 644 | {}: {} {} {} {} | True

File: operators/nick-qu.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+dt = 1e-5
+res = "ne"
+if a > b - dt and a < b + dt:
+    res = "eq"
+print("{} {} {} {}".format(a, b, dt, res))
+
+year = 2020
+res = "no"
+if year % 4 == 0 and year % 100 != 0 or year % 400 == 0:
+    res = "yes"
```
Comment (conke) [inline operators/nick-qu.py]: line 46 & 48存在重复赋值的概率（下同）

File: operators/nick-qu.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+n = 6
+res = "odd"
+if n & 1 == 0:
+    res = "even"
+print("{} {}".format(n, res))
+
+n = 16
+res = "no"
+if n & n - 1 == 0:
```
Comment (conke) [inline operators/nick-qu.py]: what if n == 0?

**Comment by Nick-Qu** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 11 22 => 22 11 | {} {} => {} {} | True
2 | 11 22 => 22 11 | {} {} => {} {} | True
3 | 11 22 => 22 11 | {} {} => {} {} | True
4 | 3.14159 3.14156 1e-05 ne | {} {} {} {} | True
5 | 2020 yes | {} {} | True
6 | 5 odd | {} {} | True
7 | 6 even | {} {} | True
8 | 16 yes | {} {} | True
9 | 16 x 17 = 272 | {} x 17 = {} | True
10 | 18 % 8 = 2 | {} % 8 = {} | True
11 | 128 / 16 = 8 | {} / 16 = {} | True
12 | 666: 600 777 666 644 | {}: {} {} {} {} | True

**Pull request merged**